### PR TITLE
EVG-2132: silent additional alerting noise

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

**Comment by tychoish** [comment]
This builds on top of #234 but is presented separately in case we want to do a deploy tomorrow morning without me.

File: alerts/project_triggers.go [hunk 1]
```diff
@@ -146,21 +152,26 @@ func (trig TaskFailTransition) ShouldExecute(ctx triggerContext) (bool, error) {
...
 			errMessage["lastAlert"] = lastAlerted
-			errMessage["outcome"] = "sending alert"
-			grip.Info(errMessage)
-			return true, nil
+			if maybeSend {
+				errMessage["outcome"] = "sending alert 25% of the time"
+			}
+			grip.Warning(errMessage)
+
+			return maybeSend, nil
```
Comment (bsamek) [inline alerts/project_triggers.go]: Why the added return? Returning from the function only 25% of the time, only when we log, seems odd.

File: alerts/project_triggers.go [hunk 2]
```diff
@@ -178,6 +189,13 @@ func getShouldExecuteError(ctx triggerContext) message.Fields {
 			"variant": ctx.task.BuildVariant,
 			"project": ctx.task.Project,
 		},
+		"previous": map[string]interface{}{
+			"id":          ctx.previousCompleted.Id,
```
Comment (bsamek) [inline alerts/project_triggers.go]: Could previousCompleted be a nil pointer?

File: alerts/triggers.go [hunk 3]
```diff
@@ -70,13 +70,16 @@ func newAlertRecord(ctx triggerContext, alertType string) *alertrecord.AlertReco
...
 	if ctx.task != nil {
-		record.TaskId = ctx.task.Id
 		record.ProjectId = ctx.task.Project
 		record.VersionId = ctx.task.Version
 		record.RevisionOrderNumber = ctx.task.RevisionOrderNumber
 		record.TaskName = ctx.task.DisplayName
 		record.Variant = ctx.task.BuildVariant
 
+		if record.TaskId == "" {
+			record.TaskId = ctx.task.Id
```
Comment (bsamek) [inline alerts/triggers.go]: Could ctx.task be a nil pointer?

Comment (tychoish) [reply alerts/triggers.go]: nope, because of ln 72

File: alerts/project_triggers.go (hunk 1 above)
Comment (tychoish) [reply alerts/project_triggers.go]: we were always returning on old line 163. 

it's clearly not correct that we send alerts every time (e.g. returning true,) but without more logging it's hard to see why we're sending alerts each time. 

clarified message

**Pull request merged**